src/emulator/modules/SceGxm/SceGxm.cpp

```diff
@@ -858,56 +858,61 @@ EXPORT(int, sceGxmProgramIsSpriteCoordUsed) {
     return UNIMPLEMENTED();
 }
 
-EXPORT(int, sceGxmProgramParameterGetArraySize) {
-    return UNIMPLEMENTED();
+EXPORT(int, sceGxmProgramParameterGetArraySize, const SceGxmProgramParameter *parameter) {
+    return parameter->array_size;
 }
 
 EXPORT(int, sceGxmProgramParameterGetCategory, const SceGxmProgramParameter *parameter) {
-    assert(parameter != nullptr);
-
     return parameter->category;
 }
 
-EXPORT(int, sceGxmProgramParameterGetComponentCount) {
-    return UNIMPLEMENTED();
+EXPORT(int, sceGxmProgramParameterGetComponentCount, const SceGxmProgramParameter *parameter) {
+    return (int32_t)parameter->component_count;
 }
 
-EXPORT(int, sceGxmProgramParameterGetContainerIndex) {
-    return UNIMPLEMENTED();
+EXPORT(int, sceGxmProgramParameterGetContainerIndex, const SceGxmProgramParameter *parameter) {
+    return parameter->container_index;
 }
 
-EXPORT(int, sceGxmProgramParameterGetIndex) {
-    return UNIMPLEMENTED();
+EXPORT(uint32_t, sceGxmProgramParameterGetIndex, const SceGxmProgram *program, const SceGxmProgramParameter *parameter) {
+    uint64_t parameter_offset = program->parameters_offset;
+
+    if (parameter_offset > 0)
+        parameter_offset += (uint64_t)&program->parameters_offset;
+    return (uint32_t)((uint64_t)parameter - parameter_offset) >> 4;
 }
 
-EXPORT(Ptr<void>, sceGxmProgramParameterGetName, Ptr<const SceGxmProgramParameter> parameter) {
-    return Ptr<void>(parameter.address() + parameter.get(host.mem)->name_offset);
+EXPORT(Ptr<const char>, sceGxmProgramParameterGetName, Ptr<const SceGxmProgramParameter> parameter) {
```

Comment:
@petmac Can this be rewritten to use raw C pointers? I'm not sure what the recent (?) bridging change was about.

Reply:
I don't think so, because there's no safe way to convert back from a raw pointer to an address (Ptr).

Reply:
I see, thanks.